pkg/pmem-csi-operator/controller/deployment/deployment_controller.go

```diff
@@ -117,14 +247,16 @@ func NewReconcileDeployment(client client.Client, opts pmemcontroller.Controller
 	evRecorder := evBroadcaster.NewRecorder(scheme.Scheme, corev1.EventSource{Component: "pmem-csi-operator"})
 
 	return &ReconcileDeployment{
-		client:         client,
-		evBroadcaster:  evBroadcaster,
-		evRecorder:     evRecorder,
-		k8sVersion:     opts.K8sVersion,
-		namespace:      opts.Namespace,
-		containerImage: opts.DriverImage,
-		deployments:    map[string]*pmemcsiv1alpha1.Deployment{},
-		reconcileHooks: map[ReconcileHook]struct{}{},
+		client:           client,
+		evBroadcaster:    evBroadcaster,
+		evRecorder:       evRecorder,
+		k8sVersion:       opts.K8sVersion,
+		namespace:        opts.Namespace,
+		containerImage:   opts.DriverImage,
+		deployments:      map[string]*pmemcsiv1alpha1.Deployment{},
+		deploymentsMutex: sync.Mutex{},
+		reconcileMutex:   sync.Mutex{},
```

Comment:
You can remove the initialization of `deploymentsMutex` and `reconcileMutex`, it's not needed.